src/mongocxx/options/server_api.hpp

```diff
@@ -0,0 +1,143 @@
+// Copyright 2021 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <mongocxx/config/prelude.hpp>
+
+#include <string>
+
+#include <bsoncxx/stdx/optional.hpp>
+#include <bsoncxx/stdx/string_view.hpp>
+#include <mongocxx/stdx.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+class client;
+class pool;
+
+namespace options {
+
+///
+/// Class representing options for server API.
+///
+class MONGOCXX_API server_api {
+   public:
+    ///
+    /// Enum representing the possible values for server API version.
+    ///
+    enum class version { k_version_1 };
```

Comment:
I suppose, if you wanted to, that you could replace `k_version_1` here with `k_1`, since you have kept it as an enum class, and `version` is already in the name. Totally up to you. At three characters, `k_1` is getting awfully short and potentially into the "what if it conflicts with some crazy system macro" territory.

Reply:
Yeah `server_api::version::k_version_1` reads a little clearer to me even if redundant. Unless @kevinAlbs has a strong opinion, I'll probably keep it as `k_version_1`.